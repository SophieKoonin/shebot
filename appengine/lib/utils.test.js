const { getAttendancePostMessage } = require('./utils');

describe('utils', () => {
  describe('getAttendancePostMessage', () => {
    it('should format correctly when there is no team updates meeting', () => {
      const notes = '';
      const mainSong = 'Bills Bills Bills';
      const runThrough = 'Poison';
      const result = getAttendancePostMessage({ mainSong, runThrough, notes });
      expect(result).toMatchSnapshot();
    });

    it('should format correctly when there is a team updates meeting', () => {
      const notes = 'blah blah Team Updates blah blah';
      const mainSong = 'Bills Bills Bills';
      const runThrough = 'Poison';
      const result = getAttendancePostMessage({ mainSong, runThrough, notes });
      expect(result).toMatchSnapshot();
    });
    it('should format correctly when there is no run through', () => {
      const mainSong = 'Bills Bills Bills';
      const notes = '';
      const result = getAttendancePostMessage({ mainSong, notes });
      expect(result).toMatchSnapshot();
    });
  });
});
