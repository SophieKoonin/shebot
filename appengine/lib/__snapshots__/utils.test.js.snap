// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils getAttendancePostMessage should format correctly when there is a team updates meeting 1`] = `
":dancing_banana: Rehearsal day! :dancing_banana: <!channel> 
*Today's rehearsal:* Bills Bills Bills
 *Run through:* Poison

 *Team updates meeting at 6:30! All welcome* :tada:
Please indicate whether or not you can attend tonight by reacting to this message with :thumbsup: (present) or :thumbsdown: (absent).
Facilitator please respond with :raised_hands:!
To volunteer for Physical warm up, respond with :muscle: For Musical warm up, respond with :musical_note:."
`;

exports[`utils getAttendancePostMessage should format correctly when there is no run through 1`] = `
":dancing_banana: Rehearsal day! :dancing_banana: <!channel> 
*Today's rehearsal:* Bills Bills Bills
  Please indicate whether or not you can attend tonight by reacting to this message with :thumbsup: (present) or :thumbsdown: (absent).
Facilitator please respond with :raised_hands:!
To volunteer for Physical warm up, respond with :muscle: For Musical warm up, respond with :musical_note:."
`;

exports[`utils getAttendancePostMessage should format correctly when there is no team updates meeting 1`] = `
":dancing_banana: Rehearsal day! :dancing_banana: <!channel> 
*Today's rehearsal:* Bills Bills Bills
 *Run through:* Poison

 Please indicate whether or not you can attend tonight by reacting to this message with :thumbsup: (present) or :thumbsdown: (absent).
Facilitator please respond with :raised_hands:!
To volunteer for Physical warm up, respond with :muscle: For Musical warm up, respond with :musical_note:."
`;
